src/Analyzers/Core/CodeFixes/AddParameter/CodeFixData.cs

```diff
@@ -8,31 +8,25 @@
 
 namespace Microsoft.CodeAnalysis.AddParameter
 {
-    internal readonly struct CodeFixData
+    internal readonly struct CodeFixData(
+        IMethodSymbol method,
+        Func<CancellationToken, Task<Solution>> createChangedSolutionNonCascading,
+        Func<CancellationToken, Task<Solution>>? createChangedSolutionCascading)
     {
-        public CodeFixData(
-            IMethodSymbol method,
-            Func<CancellationToken, Task<Solution>> createChangedSolutionNonCascading,
-            Func<CancellationToken, Task<Solution>>? createChangedSolutionCascading)
-        {
-            Method = method ?? throw new ArgumentNullException(nameof(method));
-            CreateChangedSolutionNonCascading = createChangedSolutionNonCascading ?? throw new ArgumentNullException(nameof(createChangedSolutionNonCascading));
-            CreateChangedSolutionCascading = createChangedSolutionCascading;
-        }
 
         /// <summary>
         /// The overload to fix.
         /// </summary>
-        public IMethodSymbol Method { get; }
+        public IMethodSymbol Method { get; } = method ?? throw new ArgumentNullException(nameof(method));
```

Comment:
❔ What does the default constructor look like for a struct with a primary constructor and field initializers?

Reply:
the same as when you have a normal struct with a normal constructor that assigns to the fields.